### PR TITLE
Add auth'd app tile using code exchange

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

  - Adds AppTile that can authenticate via code

## Screenshots

https://user-images.githubusercontent.com/76954025/234919620-de6723b5-2d32-4320-b6de-b13767ffa4b6.mp4

### CODE REVIEW

File: src/hooks/useExchangeToken.test.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    isFetched: true,
+  });
+  useAuthMock.mockReturnValue({
+    authResult: {
+      accessToken: 'someToken',
+    },
+  });
+});
+
+test('fetches and parses $me', async () => {
```
Comment (markdlv) [inline src/hooks/useExchangeToken.test.tsx:69]: Is this test title copy pasta?

File: src/hooks/useExchangeToken.test.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  axiosMock.onPost('/v1/client-tokens').reply<ExchangeResult>(200, {
+    code: 'some-code',
+  });
+  const { result } = await renderHookInContext();
+  await waitFor(() => result.current.isSuccess === true);
+  console.log(result.current);
+  console.log(axiosMock.history.post);
```
Comment (markdlv) [inline src/hooks/useExchangeToken.test.tsx:76]: Can remove these?

File: src/screens/AuthedAppTileScreen.test.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    },
+    clientId: 'someClientId',
+    callbackUrls: ['http://unit-test/app-tile/callback'],
+  };
+  route.params.appTile = appTile;
+  render(<AuthedAppTileScreen navigation={navigation} route={route} />);
+  expect(useExchangeTokenMock).toHaveBeenCalledTimes(1);
+  expect(webviewMock.mock.calls[0][0]).toMatchObject({
+    source: {
+      uri: 'http://unit-test/app-tile/callback?code=someCode',
```
Comment (markdlv) [inline src/screens/AuthedAppTileScreen.test.tsx:69]: Totally good to merge as-is, but I think passing `accountId` and `projectId` to the 3rd party app would be great.  We've run into this before and having these passed in would be super helpful.  If they're not used, no harm.

**Review by markdlv — DISMISSED** [review]
🔥

File: src/hooks/useExchangeToken.test.tsx (hunk 1 above)
Comment (bruce-glazier) [reply src/hooks/useExchangeToken.test.tsx:69]: Indeed!

**Review by sternetj — APPROVED** [review]

File: src/screens/AuthedAppTileScreen.test.tsx (hunk 3 above)
Comment (bruce-glazier) [reply src/screens/AuthedAppTileScreen.test.tsx:69]: Ohh, okay, that makes sense. I'll do a quick update to address these comments.

File: src/hooks/useExchangeToken.test.tsx (hunk 1 above)
Comment (bruce-glazier) [reply src/hooks/useExchangeToken.test.tsx]: Updated

File: src/hooks/useExchangeToken.test.tsx (hunk 2 above)
Comment (bruce-glazier) [reply src/hooks/useExchangeToken.test.tsx]: Done

File: src/screens/AuthedAppTileScreen.test.tsx (hunk 3 above)
Comment (bruce-glazier) [reply src/screens/AuthedAppTileScreen.test.tsx]: Updated to always include them

File: src/screens/AuthedAppTileScreen.tsx [hunk 4]
```diff
@@ -51,18 +49,15 @@ export const AuthedAppTileScreen = ({ navigation, route }: Props) => {
     }
 
     // These params will only be needed for LifeOmic app tiles
```
Comment (sternetj) [inline src/screens/AuthedAppTileScreen.tsx:51]: We probably don't need this comment anymore.

**Review by sternetj — DISMISSED** [review]

File: src/screens/AuthedAppTileScreen.tsx (hunk 4 above)
Comment (bruce-glazier) [reply src/screens/AuthedAppTileScreen.tsx]: Yup, good call. Removed.

**Review by sternetj — APPROVED** [review]

**Pull request merged**